geovin/sd-v3/src/index.ts

```diff
@@ -0,0 +1,1167 @@
+            }
+        }
+    },
+    "G21": {
+        "1007": {
+            "position": {
+                "x": -611.5780881073982,
+                "y": -2031.8101135008642,
+                "z": 909.6802762548646
+            },
+            "target": {
+                "x": 0,
+                "y": -22.60634888556615,
+                "z": 418.7894859313965
+            }
+        }
+    },
+    "G22": {
+        "1570": {
+            "position": {
+                "x": -775.4135660780335,
+                "y": -2264.820363766209,
+                "z": 893.1628044897665
+            },
+            "target": {
+                "x": 0,
+                "y": -22.60634888556615,
+                "z": 406.2894859313965
+            }
+        },
+        "1774": {
+            "position": {
+                "x": -775.4135660780335,
+                "y": -2264.820363766209,
+                "z": 893.1628044897665
+            },
+            "target": {
+                "x": 0,
+                "y": -22.60634888556615,
+                "z": 406.2894859313965
+            }
+        }
+    },
+    "G23": {
+        "1007": {
+            "position": {
+                "x": -598.1586425875126,
+                "y": -2570.151503129709,
+                "z": 1299.6844911835533
+            },
+            "target": {
+                "x": 0,
+                "y": -22.60634888556615,
+                "z": 646.7894859313965
+            }
+        }
+    }
+};
+
+class GeovinSD {
+
+    private initial_parameter_values: any = {};
+
+    private sd_ticket: string = "";
+
+    public initialized: boolean = false; // it is loaded and ready
+
+    public activated: boolean = false; // we have loaded the current settings to see
+
+    public visible: boolean = false; // we can see it
+
+    public rendering: boolean = false; // it's processing information
+
+    public movingCamera: boolean = false; // we are moving the camera
+
+    private currentSession: any = null;
+
+    private currentViewport: any = null;
+
+    public parameters: any = null;
+
+    public cameraSettings : any;
+    public activeScene: string = "";
+    public availableScenes: any;
+    public activeWidth: string = "";
+    public activeShape: string = "";
+    private isBed: boolean = false;
+    private isPainted: boolean = false;
+
+    constructor(sd_ticket: string ) {
+        this.sd_ticket = sd_ticket;
+    }
+    
+    public async init(values: any = {}) {
+        this.rendering = true;
+        this.initial_parameter_values = values;
+        $('.sd-curtain span').text('Initializing your model...');
+        $('.loading-bar__inner').css({width:'33%'});
+        $('.loading-bar__inner').text('33%');
+        
+        await this.createViewport();
+        await this.createSession( this.sd_ticket );
+        this.initialized = true;
+        //await this.addDefaultScene();
+        this.availableScenes = this.currentViewport.lightScenes;
```

Comment:
This may be a good opportunity to do conditional checks on `this.currentViewport` (i.e. `this.currentViewport?.lightScenes`), since the currentViewport is being loaded in via an async function.